itest/tests/preferences.test.js

```diff
@@ -0,0 +1,28 @@
+/* @flow */
+import path from "path"
+
+import {
+  defaultModalButton,
+  ingestProgressBar,
+  jsonTypeConfigInput,
+  viewerResults
+} from "../../src/js/test/locators"
+import {stdTest} from "../lib/jest"
+import createTestBrim from "../lib/createTestBrim"
+
+const config = path.join(__dirname, "../testdata/custom-schema.json")
+
+describe("Preferences Modal Tests", () => {
+  const brim = createTestBrim("preferences.test")
+
+  stdTest("setting a custom json type config", async (done) => {
+    brim.clickAppMenuItem("preferences")
+    await brim.setValue(jsonTypeConfigInput, config)
+    await brim.click(defaultModalButton)
+    await brim.ingest("custom-sample.ndjson")
+    await brim.waitUntil(() => brim.isNotVisible(ingestProgressBar))
+
+    expect(await brim.getText(viewerResults)).toMatchSnapshot()
```

Comment:
In general, bare, zql-less log results are prone to giving nondeterminstic results. That could be why this test failed on Windows. Can I suggest what I used in #817? It's very equivalent to the search used in the ingest tests:
```
_path=conn proto=tcp | cut ts, src_ip, src_port, dst_ip, dst_port, proto | sort ts
```
Using this gets more bang for your buck as a test: not only do you get (or are supposed to get) determinstic results, but also you exercise the custom schema (note the use of `src_ip` et al, which aren't in normal Zeek logs).

Reply:
Yes! Thanks for that.